### PR TITLE
Feature/skill configuration

### PR DESCRIPTION
**Pull request by Mn0491**
==== Fixed Issues ====
NONE

====  Tech Notes ==== 
settings.py is extended to make REST api calls to home.mycroft.ai to allow skill configurations. currently does a 60 second poll to home.mycroft.ai to pull changes skill configuration changes.  

====  Documentation Notes ==== 
Mycroft skill developers can now add a settingsmeta.json file into their skills, and the core will automatically upload the file to home.mycroft.ai for you to input the settings. This will then go back to the core and create a settings.json file that can be used for different settings in the skills.

settingsmeta.json examples:
`
{
  "identifier": "PandoraSkill",
  "name": "Pandora",
  "skillMetadata": {
      "sections": [
          {
              "name": "Login",
              "fields": [
                  {
                      "name": "email",
                      "type": "email",
                      "label": "Email",
                      "value": ""
                  },
                  {
                      "name": "password",
                      "type": "password",
                      "label": "Password",
                      "value": ""
                  }
              ]
          }
      ]
  }
}
`

==== Localization Notes ==== 

NONE

==== Environment Notes ==== 

NONE 

==== Protocol Notes ====
settings.json, settingsmeta.json are the default file names the core will be looking at inside your skills folder.

### CODE REVIEW

File: test/skills/settings.py [hunk 1]
```diff
@@ -8,68 +8,65 @@
...
 
     def test_add_value(self):
-        s = SkillSettings(join(dirname(__file__), 'settings.json'))
+        s = SkillSettings(join(dirname(__file__), 'settings'))
         s['test_val'] = 1
-
-    def test_load_existing(self):
-        s = SkillSettings(join(dirname(__file__), 'settings', 'existing.json'))
-        self.assertEqual(len(s), 4)
+        self.assertEqual(s['test_val'], 1)
```
Comment (forslund) [inline test/skills/settings.py]: This is a kind of important test case and should be implemented in some way

File: mycroft/skills/settings.py [hunk 2]
```diff
@@ -30,32 +30,45 @@
...
         SkillSettings creates a dictionary that can easily be stored
-        to file, serialized as json.
+        to file, serialized as json. It also syncs to the backend for
+        skill settings
 
         Args:
             settings_file (str): Path to storage file
     """
-    def __init__(self, settings_file):
+    def __init__(self, directory):
```
Comment (Mn0491) [inline mycroft/skills/settings.py]: resolved

File: test/skills/settings.py (hunk 1 above)
Comment (Mn0491) [inline test/skills/settings.py]: resolved

File: mycroft/skills/settings.py [hunk 3]
```diff
@@ -30,32 +30,45 @@
...
+        # set file paths
+        self._settings_path = join(directory, 'settings.json')
+        self._meta_path = join(directory, 'settingsmeta.json')
+        self._api_path = "/" + self._device_identity + "/skill"
 
         self.loaded_hash = hash(str(self))
 
-    @property
-    def _is_stored(self):
-        return hash(str(self)) == self.loaded_hash
```
Comment (Mn0491) [inline mycroft/skills/settings.py]: resolved. I put it back and used it in poll_settings

**Pull request merged**